dataactcore/models/jobTrackerInterface.py

```diff
@@ -229,6 +230,62 @@ def checkJobDependencies(self,jobId):
                 # add to the job queue
                 jobQueueResult = self.jobQueue.enqueue.delay(depJobId)
 
+    def runChecks(self,jobId):
+        """ Run all checks on this jobId
+
+        Args:
+        jobId -- ID of job to be checked
+        Returns:
+        True if all passed, otherwise False or exception
+        """
+        if(self.checkJobReady(jobId) and self.checkPrerequisites(jobId)):
+            # All passed
+            return True
+        else:
+            return False
+
+    def checkJobReady(self, jobId):
+        """ Check that the jobId is located in job table and that status is ready
+        Args:
+        jobId -- ID of job to be run
+
+        Returns:
+        True if job is ready, False otherwise
+        """
+        query = self.session.query(Job.job_status_id).filter(Job.job_id == jobId)
+        result = self.checkJobUnique(query)
+        # Found a unique job
+        if(result.job_status_id != self.getJobStatusId("ready")):
+            # Job is not ready
+            # Job manager is not yet implemented, so for now doesn't have to be ready
+            return True
+            # TODO when job manager exists, change to exception below
+            #exc = ResponseException("Job is not ready",StatusCode.CLIENT_ERROR,None,ValidationError.jobError)
+            #exc.status = 400
+            #raise exc
```

Comment:
So this is much older code that is just being moved into the core in this PR.  Actually the job manager is now implemented, but it has been useful in the past to be able to rerun validations on a finished job, so I think I will remove this check entirely.